features/comments_and_kudos/comments_adminposts.feature

```diff
@@ -35,3 +35,40 @@ Feature: Commenting on admin posts
       And all emails have been delivered
     When I edit a comment
     Then 1 email should be delivered to "narcis"
+
+  Scenario: Comment permissions on admin posts
+    Given I have posted an admin post
+      And I am logged in as a "communications" admin
+    When I go to the admin-posts page
+      And I follow "Edit"
+    Then I should see "Who can comment on this"
+      And I should see "Registered users and guests can comment"
+      And I should see "Only registered users can comment"
+      And I should see "No one can comment"
+    When I choose "No one can comment"
+      And I press "Post"
+    Then I should see "successfully updated"
+    When I am logged in as "regular"
+      And I go to the admin-posts page
+      And follow "Comment"
```

Comment:
Hmmm... This is the news posts listing (https://test.archiveofourown.org/admin_posts) where we show the comment button on every post.  The individual posts (e.g. https://test.archiveofourown.org/admin_posts/1) hide the comment button correctly.  I'll see what I can do about this.

Reply:
Something like this?

<img width="1440" alt="AO3-6027_admin_post_index" src="https://user-images.githubusercontent.com/6746746/99477745-236df380-2921-11eb-8e83-4499e35524c5.png">

Reply:
I think for both "no comments allowed" and "only users can comment" (in case you're logged out), we can skip showing the button entirely. I'd wait for @sarken's review, though.

Reply:
+1 on skipping it entirely

Reply:
Might that be confusing to users who can't tell why the button is missing for some posts and cause them to think something isn't working?

Reply:
I think I'd rather try our standard approach of omitting the button entirely for now, and if it causes confusion, then we can display the full blue notice banner at the bottom of the post content, to make it a closer match to what you should see on the admin_post show page. (In my ideal world, the news posts on the admin_posts index would be truncated and we wouldn't even have that "Add Comment" button visible until you were on the individual admin_post show page with the full post content, but that's a bit outside the scope here.)

Reply:
I'm already modifying that view.  If we want to remove the comment button from the index entirely, that's actually simpler than putting in logic to selectively show the button depending on permissions.  It would then force users to click through to the individual post to comment, where the message will be obvious.

Reply:
Eh, might as well, then! (I don't think there's as much justification for removing it when we're displaying the full text of the post, but I also... don't really care that much about justifying it.)

Reply:
Should I make a separate task for that ideal case?  Making the admin_posts index behave more like the news module on the home page with "read more" links?